cloudfoundry/provider/types_app.go

```diff
@@ -728,13 +784,42 @@ func setEnvForUpdate(ctx context.Context, existingEnvs basetypes.MapValue, plann
 
 	finalEnvs = make(map[string]*string)
 
-	diagnostics.Append(existingEnvs.ElementsAs(ctx, &oldEnvs, false)...)
-	diagnostics.Append(plannedEnvs.ElementsAs(ctx, &newEnvs, false)...)
+	// Handle existing environment variables
```

Comment:
I do not understand the extensive refactoring of the code here that is not related to the intent of the PR focusing on lifecycle (or am I missing something.